woocommerce/woocommerce
# Allow passing partial data to Store API extensions field.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This API makes it possible to pass partial data to Store API extensions field. Right now, anyone consuming Store API and passing back extensions data would get a 400 because they're missing a order-attribution key. This fixes it.

This also means that for Store API, extensions data is now optional, it always been optional, but only if you omit everything, now it's optional per field.


### How to test the changes in this Pull Request:

1. From a REST client, build up a cart by `GET /wc/store/cart` to get a nonce, `POST /wc/store/cart/add-item` with a product id in the body to get a Cart-Token header, and then `POST /wc/store/checkout` with the following body.
```json
{
	"shipping_address": {
		"first_name": "Name",
		"last_name": "Last Name",
		"company": "",
		"address_1": "Address",
		"address_2": "",
		"city": "SF",
		"state": "CA",
		"postcode": "94000",
		"country": "US",
		"phone": "055334432"
	},
	"billing_address": {
		"first_name": "nadir",
		"last_name": "Seghir",
		"company": "",
		"address_1": "Address",
		"address_2": "",
		"city": "SF",
		"state": "CA",
		"postcode": "94000",
		"country": "US",
		"email": "hello@gmail.co",
		"phone": "055334432"
	},
	"payment_method": "cod",
	"extensions": {
		"my-key": {
			"my-data": false
		}
	}
}
```
3. Order should pass fine.
